mmv1/third_party/terraform/tests/resource_spanner_database_test.go.erb

```diff
@@ -390,6 +395,83 @@ resource "google_spanner_database" "basic" {
 `, instanceName, instanceName, databaseName, databaseName, databaseName)
 }
 
+func TestAccSpannerDatabase_enableDropProtection(t *testing.T) {
+	t.Parallel()
+
+	rnd := RandString(t, 10)
+	instanceName := fmt.Sprintf("tf-test-%s", rnd)
+	databaseName := fmt.Sprintf("tfgen_%s", rnd)
+
+	VcrTest(t, resource.TestCase{
+		PreCheck:                 func() { acctest.AccTestPreCheck(t) },
+		ProtoV5ProviderFactories: ProtoV5ProviderFactories(t),
+		CheckDestroy:             testAccCheckSpannerDatabaseDestroyProducer(t),
+		Steps: []resource.TestStep{
+			{
+				// Test creating a database with `enable_drop_protection` set
+				Config: testAccSpannerDatabase_enableDropProtection(instanceName, databaseName),
+				Check: resource.ComposeTestCheckFunc(
+					resource.TestCheckResourceAttrSet("google_spanner_database.basic", "state"),
+					resource.TestCheckResourceAttr("google_spanner_database.basic", "enable_drop_protection", "true"),
+          // reset enable_drop_protection value to false for cleanup
+          testAccResetEnableDropDatabaseProtection(t),
```

Comment:
IIUC, we shouldn't be cleaning up the resource until after the update step `testAccSpannerDatabase_enableDropProtectionUpdate` and since we update `enable_drop_protection` to false, we shouldn't need this reset and the test run into error without this reset, it may indicate that the update of this field is not working?